### PR TITLE
Update glossary.md

### PR DESCRIPTION
**Pull request by stew**
add a definition of the nix database to the glossary

### CODE REVIEW

File: doc/manual/src/glossary.md [hunk 1]
```diff
@@ -34,6 +34,10 @@
     the same output. This cannot be guaranteed in general (e.g., a
     builder can rely on external inputs such as the network or the
     system time) but the Nix model assumes it.
+    
+  - Nix database\
+    a sqlite database in contained in the Nix store which contains metadata
```
Comment (edolstra) [inline doc/manual/src/glossary.md:39]: "in contained in" is not grammatical. Also, the DB is not *in* the Nix store but elsewhere (/nix/var/nix/db).

"a" should be capitalized. "sqlite" should be "SQLite".

**Review by edolstra — CHANGES_REQUESTED** [review]

File: doc/manual/src/glossary.md (hunk 1 above)
Comment (fricklerhandwerk) [reply doc/manual/src/glossary.md:39]:
```suggestion
    An SQlite database to track references between store objects. Default location: `/nix/var/nix/db`.
```

**Comment by fricklerhandwerk** [comment]
@stew thanks for adding the info. Fixed up grammar as requested by @edolstra, added anchors and links for better discoverability.

File: doc/manual/src/glossary.md [hunk 2]
```diff
@@ -86,6 +86,13 @@
     the same output. This cannot be guaranteed in general (e.g., a
     builder can rely on external inputs such as the network or the
     system time) but the Nix model assumes it.
+    
+  - Nix database{#gloss-nix-database}\
+    An SQlite database to track references between store objects.
+    Default location: `/nix/var/nix/db`.
+    about the Nix store such as hashes and references.
```
Comment (fricklerhandwerk) [inline doc/manual/src/glossary.md:93]:
```suggestion
```

Artifact

File: doc/manual/src/glossary.md [hunk 3]
```diff
@@ -86,6 +86,13 @@
     the same output. This cannot be guaranteed in general (e.g., a
     builder can rely on external inputs such as the network or the
     system time) but the Nix model assumes it.
+    
+  - Nix database{#gloss-nix-database}\
+    An SQlite database to track references between store objects.
```
Comment (fricklerhandwerk) [inline doc/manual/src/glossary.md:91]:
```suggestion
    An SQlite database to track references between store objects in a [local store][store].
```

This is an implementation detail of the local store and that should at least be made clear.

**Comment by fricklerhandwerk** [comment]
continued as #8167

**Pull request closed without merging**